stock_available_to_promise_release/models/stock_move.py

```
@@ -622,6 +670,30 @@ def unrelease(self, safe_unrelease=False):
                     origin_moves.write({"propagate_cancel": False})
                     # origin_moves._action_cancel()
                     moves_to_cancel |= origin_moves
+                # checking that for the current step (pick/pack/ship)
+                # move.product_uom_qty == step.cancelled_qty + move.returned_quanty
+                # If not the case, we have to move back goods in stock.
+                qty_cancelled_at_step = sum(origin_moves.mapped("product_uom_qty"))
+                qty_to_return_at_step = (
+                    qty_to_unrelease - qty_cancelled_at_step - qty_returned_for_move
+                )
+                # TODO check move.rule_id.allow_unrelease_on_cancel ?
```

No need before the if nothing to return
but after I would raise an error if we need to return but the option is not enabled

---

Well, we need to have this compatible with both
If we have 2 routes one with setting enabled, one with setting disabled, do we want to raise an exception?
To me we shouldn't raise an exception, but do nothing if setting is disabled.
What do you think?

---

If you multiple routes, only one is taken into consideration. There is only one rule_id on the move. Only that one matters

---

if the setting is disabled, then you should not reach the part where you need to return a move. This was already checked beforehand in the check method. Nevertheless, it is good to raise same error here for safety